### PR TITLE
chore(lint): Run prettier on most files and move ABIs to own file

### PR DESCRIPTION
**Pull request by CoreyMartin**
- ABIs are difficult to conform to prettier and are referenced in multiple files so they've been moved to a prettier ignored file
- Run prettier on all non conforming files
- Add `prettier --check .` as a test requirement in CI so that PRs must conform before merge

### CODE REVIEW

**Comment by jasonstockman** [comment]
Was just thinking about this a few minutes ago! Moving ABI to their own file sounds good. For the prettifier we should run that for sure on everything at some point, but probably wait to do it once all pending PR are done, otherwise it will make merges tough. I'd like to also propose we `"printWidth": 120` in the prettier config so we have less forced line wraps in functions.

**Comment by CoreyMartin** [comment]
@jasonstockman Sounds good! Updated printWidth. I checked the merge status against CurveTests branch and it's actually not too bad, 3 file conflicts. I can totally wait to merge this until after yours is merged and fix the conflicts here. Since this PR is adding the check to the tests the files need to pass in order for the merge to happen.

File: MainnetHelper.js
Comment (yotamdaniel) [inline MainnetHelper.js:1]: Duplicate file?

File: test/VaultOUSD.js
Comment (yotamdaniel) [inline test/VaultOUSD.js:44]: can this be wrapped to a single line?

File: .github/workflows/BuildAndTest.yml
Comment (jasonstockman) [inline .github/workflows/BuildAndTest.yml:26]: Should these also be v3?

**Review by jasonstockman — APPROVED** [review]

**Comment by jasonstockman** [comment]
> 

No need to wait on my changes. I can handle those once this is approved. The CurveTests branch wont be ready today and I'd rather get this in ASAP.

File: .github/workflows/BuildAndTest.yml
Comment (CoreyMartin) [reply .github/workflows/BuildAndTest.yml:26]: yeah these are v3 after merging main back in

File: .github/workflows/BuildAndTest.yml
Comment (CoreyMartin) [reply .github/workflows/BuildAndTest.yml:31]: yeah these are v3 after merging main back in

**Pull request merged**
